libs/search-widget/src/core/api.ts

```diff
@@ -212,11 +212,16 @@ export const hasAuthData = (): boolean => {
 };
 
 export const getFileUrls = (paths: string[]): Observable<string[]> => {
-  if (paths.length === 0 || !isPrivateKnowledgeBox()) {
-    return of(paths.map((path) => `${getRegionalBackend()}${path}`));
-  } else {
-    return getTempToken().pipe(
-      map((token) => paths.map((path) => `${getRegionalBackend()}${path}?eph-token=${token}`)),
-    );
-  }
+  return (paths.length === 0 || !isPrivateKnowledgeBox() ? of('') : getTempToken()).pipe(
```

Comment:
I don't get this: how `return (paths.length === 0 ||…` can return an observable if `paths.length !== 0`?

Reply:
ok, I get it now. It took me some time though, which is an indicator that this one liner might deserve to be a two liners?

Reply:
because I want to write only one pipe